File: rb101_rb109/easy9/uppercase_check.rb
=begin
input: a string 
output: boolean
rules: true if all alphabetic chars are uppercase; false otherwise
empty string returns true
ignore non alphabetic chars
algorithm:
return true for empty str
count if there's more than 1 lowercase char
=end

# first attempt 

def uppercase?(str)
  return true if str.size == 0
  str.count("a-z") == 0 ? true : false
end

# another solution
=begin
def uppercase?(str)
  str == str.upcase
end
=end


puts uppercase?("")
puts uppercase?("t")
puts uppercase?("T")
puts uppercase?('Four Score')
puts uppercase?('FOUR SCORE')
puts uppercase?('4SCORE!')

File: ls215/js210_string_processing/uppercase_check.js
/*
Question: a function that checks if all alphabetic chars are uppercase
  - arg: a string
  - return: true / false
  - ignore non-alphabetic chars
  - empty string returns true
Algo:
  - create another string which is uppercase version of given arg.
  - compare the 2nd string to given arg.
*/

function isUppercase(string) {
  let allUppers = string.toUpperCase();
  return string === allUppers;
}


console.log(isUppercase('t'));
console.log(isUppercase('T'));
console.log(isUppercase('Four Score'));
console.log(isUppercase('FOUR SCORE'));
console.log(isUppercase('4SCORE!'));
console.log(isUppercase(''));    //true  